### PR TITLE
Reorg of "mlflow.store" module and splitting "tracking" metadata and "artifact" repository submodules

### PR DESCRIPTION
**Pull request by mparkhe**
## What changes are proposed in this pull request?
 
Simple code reorg

### Old structure
```
store/
├── cli.py
├── db
│   └── utils.py
├── db_migrations
│   ├── env.py
│   └── versions
│       ├── ...py
│       └── ...py
├── dbfs_artifact_repo.py
├── dbmodels
│   ├── db_types.py
│   ├── initial_models.py
│   └── models.py
│
│
├── abstract_store.py
├── file_store.py
├── rest_store.py
├── sqlalchemy_store.py
│
├── artifact_repo.py
├── artifact_repository_registry.py
├── azure_blob_artifact_repo.py
├── ftp_artifact_repo.py
├── gcs_artifact_repo.py
├── hdfs_artifact_repo.py
├── local_artifact_repo.py
├── runs_artifact_repo.py
├── s3_artifact_repo.py
└── sftp_artifact_repo.py
```

### New structure

```
store/
├── db_types.py
├── artifact
│   ├── cli.py
│   ├── artifact_repo.py
│   ├── artifact_repository_registry.py
│   ├── azure_blob_artifact_repo.py
│   ├── dbfs_artifact_repo.py
│   ├── ..._artifact_repo.py
│   └── ..._artifact_repo.py
├── db_migrations
│   ├── env.py
│   └── versions
│       ├── ...py
│       ├── ...py
│       └── ...py
├── tracking
│   ├── abstract_store.py
│   ├── dbmodels
│   │   ├── initial_models.py
│   │   └── models.py
│   ├── file_store.py
│   ├── rest_store.py
│   └── sqlalchemy_store.py
└── utils
    └── db_utils.py


```
 
## How is this patch tested?
 
(Details)
 
### Is this a user-facing change? 

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)
 
### What component(s) does this PR affect?
 
- [ ] UI
- [ ] CLI 
- [ ] API 
- [ ] REST-API 
- [ ] Examples 
- [ ] Docs
- [ ] Tracking
- [ ] Projects 
- [ ] Artifacts 
- [ ] Models 
- [ ] Scoring 
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:
 
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/store/tracking/__init__.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+Several constants are used by multiple backend store implementations.
+"""
+
+# Path to default location for backend when using local FileStore or ArtifactStore.
+# Also used as default location for artifacts, when not provided, in non local file based backends
+# (eg MySQL)
+DEFAULT_LOCAL_FILE_AND_ARTIFACT_PATH = "./mlruns"
+SEARCH_MAX_RESULTS_DEFAULT = 1000
+SEARCH_MAX_RESULTS_THRESHOLD = 50000
```
Comment (smurching) [inline mlflow/store/tracking/__init__.py]:
```suggestion
SEARCH_MAX_RESULTS_THRESHOLD = 50000

```

**Review by sueann — COMMENTED** [review]
looks generally good to me - a few questions for you inline.

File: mlflow/store/db_migrations/env.py [hunk 2]
```diff
@@ -18,7 +18,7 @@
 # for 'autogenerate' support
 # from myapp import mymodel
 # target_metadata = mymodel.Base.metadata
-from mlflow.store.dbmodels.models import Base
+from mlflow.store.tracking.dbmodels.models import Base
```
[the remark this answers is not in the document]
Comment (mparkhe) [reply mlflow/store/db_migrations/env.py]: moved this up to `store.db` module

File: mlflow/store/artifact/dbfs_artifact_repo.py [hunk 3]
```diff
@@ -4,9 +4,9 @@
 
 from mlflow.entities import FileInfo
 from mlflow.exceptions import MlflowException
-from mlflow.store.artifact_repo import ArtifactRepository
-from mlflow.store.rest_store import RestStore
-from mlflow.store.local_artifact_repo import LocalArtifactRepository
+from mlflow.store.artifact.artifact_repo import ArtifactRepository
+from mlflow.store.tracking.rest_store import RestStore
```
[the remark this answers is not in the document]
Comment (mparkhe) [reply mlflow/store/artifact/dbfs_artifact_repo.py]: IntelliJ has a brain! Does what it likes...
Tried some local ordering.

**Review by sueann — APPROVED** [review]
lgtm!

**Pull request merged**